crates/shark-lex/src/lexer.rs

```diff
@@ -41,20 +43,14 @@ impl<'a> Lexer<'a> {
             self.expected_token
                 .clone()
                 .expect("Failed to get expected token type"),
-            self.working_row_col.0,
-            self.working_row_col.1,
+            self.working_position,
             self.working_content.len(),
         ));
         self.reset_token();
     }
 
-    pub fn push_single_char_token(&mut self, kind: TokenKind, tokens: &mut Vec<LexerToken>) {
-        tokens.push(LexerToken::new(
-            kind,
-            self.current_row_col.0,
-            self.current_row_col.1,
-            1,
-        ));
+    pub fn push_single_char_token(&mut self, kind: TokenKind, tokens: &mut Vec<LexerToken<'a>>) {
+        tokens.push(LexerToken::new(kind, self.current_position, 1));
```

Comment:
The `current_position` variable should contain a path because tokens are pushed with it, and having a path associated with them is good for errors.